CRM/Utils/System.php

```diff
@@ -433,17 +433,25 @@ public static function getClassName($object) {
    *
    * @param string $url
    *   The URL to provide to the browser via the Location header.
+   * @param array $context
+   *   Optional additional information for the hook.
    */
-  public static function redirect($url = NULL) {
+  public static function redirect($url = NULL, $context = []) {
     if (!$url) {
       $url = self::url('civicrm/dashboard', 'reset=1');
     }
     // replace the &amp; characters with &
     // this is kinda hackish but not sure how to do it right
     $url = str_replace('&amp;', '&', $url);
 
+    $output = CRM_Utils_Array::value('snippet', $_GET);
+
+    $parsedUrl = CRM_Utils_Url::parseUrl($url);
+    CRM_Utils_Hook::alterRedirect($parsedUrl, $context, $output);
```

Comment:
@eileenmcnaughton , this doesn't seem match-up with the function signature:

```php
  public static function alterRedirect($url, &$outputFormat, $context) { 
```

1. Which order seems more correct?
2. What's an example where the `outputFormat` matters or needs to be altered?
3. Could we treat the `outputFormat` as part of `$context`?

Reply:
I think output format could be part of context

Reply:
OK @totten have fixed to be one array